File: sort/insertionSort.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> insertionSort(vector<int> arr){
    int temp;
    
    for(int i = 0; i < arr.size(); i++){
        for(int j = i+1; j > 0; j--){
            if(arr[j] < arr[j-1]){
                temp = arr[j];
                arr[j] = arr[j-1];
                arr[j-1] = temp;
            }
        }
    }
    return arr;
}

int main(){

    vector<int> arr = {9, 4, 3, 10, 5, 8, 7, 6, 2, 1};
    

    for(int i = 0; i < arr.size(); i++){
        cout << arr[i] << " ";
    }
    cout << endl;

    arr = insertionSort(arr);

    for(int i = 0; i < arr.size(); i++){
        cout << arr[i] << " ";
    }
    cout << endl;



    return 0;
}

File: sort/insertionSort.js

const insertionSort = (arr) => {
    const result = [...arr];
    let temp;
    
    for(let  i = 0; i < result.length; i++){
        for(let j = i+1; j >0; j--){
            if(result[j] < result[j-1]){
                temp = result[j];
                result[j] = result[j-1];
                result[j-1] = temp;
            }
        }
    }
    
    return result;
}


const arr = [9, 4, 3, 10, 5, 8, 7, 6, 2, 1];
console.log(arr);
const newArr = insertionSort(arr);
console.log(newArr);




 